transformations/noun_compound_paraphraser/README.md

```diff
@@ -0,0 +1,124 @@
+# English Noun Compound Paraphraser [N+N]→ 🐍
+This transformation replaces two-word noun compounds with a paraphrase, based
+on the compound paraphrase dataset from SemEval 2013 Task 4 (Hendrickx et al., 2013).
+This currently only works for English.
+
+Author: Juan Diego Rodriguez (juand-r@utexas.edu) <br>  Affiliation: UT Austin
+
+## What does this transformation do?
+Any two-word compound that appears in a dataset of noun compound paraphrases
+will be replaced by a paraphrase. If more than one two-word compound appears,
+then all combinations of compound paraphrases (including no paraphrase at all)
+will be returned. For example, the outputs of "After the bike ride, John stopped by the club house."
+are:
+
+```
+['After the ride on a bike, John stopped by the club house.'],
+'After the bike ride, John stopped by the house for club activities.',
+'After the ride on a bike, John stopped by the house for club activities.']
+```
+
+The number of paraphrases to be returned for each noun compound is specified by the `max_paraphrases`
+parameter (set to 1 by default). For example, the paraphrases of "club house"
+include "house for club activities", "house for club members", "house in which a club meets", etc.
+
+Currently this uses the dataset from SemEval 2013 Task 4 (Hendrickx et al., 2013),
+which has a score for each paraphrase (frequency of annotation). We start with
+replacing paraphrases with the highest score, and paraphrases with the same
+score (ties) are sorted randomly.
+
+## Why is this transformation important?
+This transformation can be used to add diversity to a dataset by rewriting
+noun compounds.
+
+## What tasks does it intend to benefit?
+This transformation could be used for any task, as a way to augment a dataset
+by rewriting noun compounds as short phrases.
+
+## Data and code provenance
+We wrote the code for this transformation. The dictionary of noun compound
+paraphrases is from SemEval 2013 Task 4 (Hendrickx et al., 2013). We used the
+union of the trial (train) and test sets, but we made the following modifications:
+
+- in cases where the same compound appeared in both the train and test set, we
+added together their frequency in order to determine the overall paraphrase score.
+- determiners at the beginning of the paraphrase were removed.
+
+## What are the limitations of this transformation?
+
+This transformation currently only checks for noun compounds from
+(Hendrickx et al., 2013) and therefore has low coverage.
```

Comment:
What happens if the phrase entered by the user doesn't have any noun compound combination?

Reply:
If there are no noun compound combination, it will return the original sentence, without modifications. I will put this in the README to make it more precise.

Reply:
Ok, it works.